src/edu/buffalo/cse/jive/finiteStateMachine/models/Context.java

```diff
@@ -18,13 +18,25 @@
 	private State currentState;
 	private State nextState;
 	private Map<State, Set<State>> states;
+	private boolean useMarker;

```

Comment:
The boolean useMarker determines if we should evaluate the shortest path for EExpression or not.
If E exists on the top level of given properties, the value should be set to true and the shortest path will be calculated.